### PR TITLE
Python3 compatibility for get_dirty_fields

### PR DESCRIPTION
**Pull request by brennerm**

### CODE REVIEW

**Comment by romgar** [comment]
For that kind of compatibility problem, it could be more interesting to use something like `six`, that is used for python 2/3 compatibility (https://pythonhosted.org/six/)

File: src/dirtyfields/dirtyfields.py [hunk 1]
```diff
@@ -28,7 +28,14 @@ def get_dirty_fields(self, check_relationship=False):
         new_state = self._as_dict(check_relationship)
         all_modify_field = {}
 
-        for key, value in new_state.iteritems():
+        # Python3
+        try:
+            items = new_state.items()
+        # Python2
+        except AttributeError:
+            items = new_state.iteritems()
```
Comment (romgar) [inline src/dirtyfields/dirtyfields.py]: Doing this Error catching seems not to be the best solution, as some libraries, like `six` (https://pythonhosted.org/six/) have been created for this kind of problem.

**Comment by brennerm** [comment]
Seems like a bit too much for me, just to fix this small issue.

File: src/dirtyfields/dirtyfields.py (hunk 1 above)
Comment (brennerm) [inline src/dirtyfields/dirtyfields.py]: Seems like a bit too much for me, just to fix this small issue.

**Comment by romgar** [comment]
It's a small issue for now. As it's a compatibility issue, I think it's better to solve it on a proper compatibility way.

**Comment by brennerm** [comment]
I'm ok with that as well. You could think about using 2to3 instead of six. It's part of the standard lib, so no additional dependency.

**Comment by romgar** [comment]
I'm on the other hand thinking of this `iteritems` stuff. Interesting when we are dealing with LOT of data, I doubt we will have a lot on django models.

**Comment by brennerm** [comment]
Do you mean to use
```python
for key, value in new_state.items():
```
for both versions?

**Comment by romgar** [comment]
Yep.

**Comment by brennerm** [comment]
Seems suitable for me as well. You rarely see django models with more than 10 fields.

**Comment by smn** [comment]
Yeah, :+1: on just using `.items()` instead

**Comment by smn** [comment]
But once this lands let's add py3 to the travis builds, possibly allowing them to fail while we investigate what's required to make it fully compatible.

**Comment by brennerm** [comment]
updated PR

**Pull request merged**